cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java

```diff
@@ -111,37 +112,88 @@ private static AmazonDynamoDB defaultDynamoDbClient() {
                    .build();
     }
 
-    private static Optional<String> getCristinIdentifier(Publication publication) {
-        return publication.getAdditionalIdentifiers()
-                   .stream()
-                   .filter(CristinEntryEventConsumer::isCristinAdditionalIdentifier)
-                   .findFirst()
-                   .map(AdditionalIdentifier::getValue);
+    private Publication persistConversionReports(PublicationRepresentations publicationRepresentations) {
+        persistCristinIdentifierInFileNamedWithPublicationIdentifier(publicationRepresentations);
+        persistPartOfCristinIdentifierIfPartOfExists(publicationRepresentations);
+        return publicationRepresentations.getPublication();
     }
 
-    private static boolean isCristinAdditionalIdentifier(AdditionalIdentifier additionalIdentifier) {
-        return CristinObject.IDENTIFIER_ORIGIN.equals(additionalIdentifier.getSource());
+    private PublicationRepresentations mapEventToNvaPublication(
+        FileContentsEvent<JsonNode> eventbody) {
+        var cristinObject = parseCristinObject(eventbody);
+        var publication = cristinObject.toPublication();
+        return new PublicationRepresentations(cristinObject, publication, eventbody);
     }
 
-    private Publication persistCristinIdentifierInFileNamedWithPublicationIdentifier(Publication publication,
-                                                                                     FileContentsEvent<JsonNode> eventBody) {
-        var cristinIdentifier =
-            getCristinIdentifier(publication);
-        var fileUri = constructSuccessFileUri(eventBody, publication);
+    private void persistPartOfCristinIdentifierIfPartOfExists(
+        PublicationRepresentations publicationRepresentations) {
+        if (cristinObjectIsPartOfAnotherPublication(publicationRepresentations.getCristinObject())) {
+            persistPartOfCristinIdentifierWithPublicationId(publicationRepresentations);
+        }
+    }
+
+    private boolean cristinObjectIsPartOfAnotherPublication(CristinObject cristinObject) {
```

Comment:
I'd be tempted to move this to the PublicationRepresentations class

Reply:
fikset